client/src/views/pages/test/ReportTestResult.js

```diff
@@ -0,0 +1,334 @@
+                    value: yup.string().required("Select a type of test."),
+                })
+                .nullable("Select a test result.")
+                .required("Select a test result."),
+            dateOfTest: yup.date().required("Enter a date of test.").typeError("Enter a date of test."),
+            address: yup
+                .string()
+                .required("Enter an address.")
+                .max(100, "Address must be 100 characters long or less."),
+            addressLine2: yup.string().max(100, "Address must be 100 characters long or less."),
+            postalCode: yup.string().required("Enter a valid postal code."),
+            province: yup
+                .object({
+                    label: yup.string().required("Select a province."),
+                    value: yup.string().required("Select a province."),
+                })
+                .nullable("Select a province.")
+                .required("Select a province."),
+        })
+        .required();
+
+    // ** Hooks
+    const {
+        control,
+        handleSubmit,
+        reset,
+        formState: { errors },
+    } = useForm({
+        defaultValues,
+        resolver: yupResolver(ReportTestSchema),
+    });
+
+    const testResultOptions = [
+        { value: "NEGATIVE", label: "NEGATIVE" },
+        { value: "POSITIVE", label: "POSITIVE" },
+    ];
+
+    const typeOfTestOptions = [
+        { value: "PCR", label: "PCR" },
+        { value: "ANTIGEN", label: "ANTIGEN" },
+    ];
+
+    const provinceOptions = [
+        { value: "Alberta", label: "Alberta" },
+        { value: "British Columbia", label: "British Columbia" },
+        { value: "Manitoba", label: "Manitoba" },
+        { value: "New Brunswick", label: "New Brunswick" },
+        { value: "Newfoundland and Labrador", label: "Newfoundland and Labrador" },
+        { value: "Northwest Territories", label: "Northwest Territories" },
+        { value: "Nova Scotia", label: "Nova Scotia" },
+        { value: "Nunavut", label: "Nunavut" },
+        { value: "Ontario", label: "Ontario" },
+        { value: "Prince Edward Island", label: "Prince Edward Island" },
+        { value: "Quebec", label: "Quebec" },
+        { value: "Saskatshewan", label: "Saskatshewan" },
+        { value: "Yukon", label: "Yukon" },
+    ];
+
+    console.log(errors);
+
+    const { patientId } = useParams();
+
+    const onSubmit = async (data) => {
+        try {
+            await submitTestResult(data, patientId, token);
+            toast.success("Test Result Submitted", {
+                position: "top-right",
+                autoClose: 5000,
+            });
+        } catch (error) {
+            toast.error("Test Result Could Not Be Submitted", {
+                position: "top-right",
+                autoClose: 5000,
+            });
+        }
+
+        reset();
+    };
+
+    return (
+        <div>
+            <BreadCrumbsPage
+                breadCrumbTitle="Add Test Result for"
```

Comment:
You can put the patient id instead of the name here to save an api call

```suggestion
                breadCrumbTitle=`Add Test Result for ${patientId}`
```